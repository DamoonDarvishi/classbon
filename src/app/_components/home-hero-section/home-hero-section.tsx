import Image from "next/image";
import { Button } from "../button";
import { IconArrowRight } from "../icons/icons";

export const HomeHeroSection: React.FC = () => {
  return (
    <section className="bg-hero-pattern bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left">
      <div className="container flex flex-col-reverse items-center xl:flex-row">
        <div className="flex flex-col gap-5 mt-12 pb-5 text-center xl:text-left">
          <h3 className="text-xl dark:text-info xl-text-2xl">Welcome to ...</h3>
          <h1 className="text-3xl font-black gradient lg:text-2xl xl:text-3xl">
            The Path to Progressing Through Programming Courses
          </h1>
          <p className="text-lg font-medium leading-8">
            No matter where you are in your programming journey, with the
            support of experienced Classbon instructors, you can climb to higher
            peaks without limitations. We always have your back.
          </p>
          <div className="justify-center xl:justify-start mt-5 flex gap-4">
            <Button variant="primary" size="normal">
              Frontend Courses <IconArrowRight fill="currentColor" />
            </Button>
            <Button variant="neutral" size="normal">
              Programming Consultation{" "}
            </Button>
          </div>
          <Image
            src="/images/frameworks.png"
            className="grayscale mt-4 opacity-70 m-auto xl:m-0"
            width={412}
            height={39}
            alt="online class"
          />
        </div>
        <Image
          src="/images/programmer-landing.svg"
          width={702}
          height={521}
          alt="online class"
        />
      </div>
    </section>
  );
};
