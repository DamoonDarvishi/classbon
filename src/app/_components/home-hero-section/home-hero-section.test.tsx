import { render } from "@testing-library/react";
import { HomeHeroSection } from "./home-hero-section";

describe("Home hero section", () => {
  test("display the home hero section", () => {
    const { container } = render(<HomeHeroSection />);
    expect(container).toMatchSnapshot(`
        <section className='bg-hero-pattern bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left'>
            <div className="container flex flex-col-reverse items-center xl:flex-row">
                <div className="flex flex-col gap-5 mt-12 pb-5 text-center xl:text-left">
                    <h3 className="text-xl dark:text-info xl-text-2xl">خوش اومدی به ...</h3>
                    <h1 className="text-3xl font-black gradient lg:text-3xl xl:text-5xl">مسیر صعود به دوره های برنامه نویسی</h1>
                    <p className="text-lg font-bold leading-8">
                    هر جای مسیر برنامه نویسی که باشی،
                    با همراهی استادهای باتجربه کلاسبن میتونی بدون محدودیت به
                    قله های بالاتر صعود کنی. ما همیشه هواتو داریم.
                    </p>
                    <div className="mt-5 flex gap-4">
                    <Button variant="accent" size="large">دوره های ری اکت و نکست</Button>
                    <Button variant="neutral" size="large">دوره های ری اکت و نکست</Button>
                    </div>
                    <Image src="/images/frameworks.png" className="grayscale mt-4 opacity-70 m-auto xl:m-0" width={412} height={39} alt="online class" />
                </div>
                <Image src="/images/programmer-landing.svg" width={702} height={521} alt="online class" />
            </div>
        </section>
        `);
  });
});
