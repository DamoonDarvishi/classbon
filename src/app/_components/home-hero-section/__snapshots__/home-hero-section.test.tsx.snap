// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home hero section display the home hero section: 
        <section className='bg-hero-pattern bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left'>
            <div className="container flex flex-col-reverse items-center xl:flex-row">
                <div className="flex flex-col gap-5 mt-12 pb-5 text-center xl:text-left">
                    <h3 className="text-xl dark:text-info xl-text-2xl">خوش اومدی به ...</h3>
                    <h1 className="text-3xl font-black gradient lg:text-3xl xl:text-5xl">مسیر صعود به دوره های برنامه نویسی</h1>
                    <p className="text-lg font-bold leading-8">
                    هر جای مسیر برنامه نویسی که باشی،
                    با همراهی استادهای باتجربه کلاسبن میتونی بدون محدودیت به
                    قله های بالاتر صعود کنی. ما همیشه هواتو داریم.
                    </p>
                    <div className="mt-5 flex gap-4">
                    <Button variant="accent" size="large">دوره های ری اکت و نکست</Button>
                    <Button variant="neutral" size="large">دوره های ری اکت و نکست</Button>
                    </div>
                    <Image src="/images/frameworks.png" className="grayscale mt-4 opacity-70 m-auto xl:m-0" width={412} height={39} alt="online class" />
                </div>
                <Image src="/images/programmer-landing.svg" width={702} height={521} alt="online class" />
            </div>
        </section>
         1`] = `
<div>
  <section
    class="bg-hero-pattern bg-no-repeat bg-center mt-5 xl:mt-20 xl:bg-left"
  >
    <div
      class="container flex flex-col-reverse items-center xl:flex-row"
    >
      <div
        class="flex flex-col gap-5 mt-12 pb-5 text-center xl:text-left"
      >
        <h3
          class="text-xl dark:text-info xl-text-2xl"
        >
          خوش اومدی به ...
        </h3>
        <h1
          class="text-3xl font-black gradient lg:text-3xl xl:text-5xl"
        >
          مسیر صعود به دوره های برنامه نویسی
        </h1>
        <p
          class="text-lg font-bold leading-8"
        >
          هر جای مسیر برنامه نویسی که باشی، با همراهی استادهای باتجربه کلاسبن میتونی بدون محدودیت به قله های بالاتر صعود کنی. ما همیشه هواتو داریم.
        </p>
        <div
          class="mt-5 flex gap-4"
        >
          <button
            class="btn btn-accent btn-lg "
            type="button"
          >
            دوره های ری اکت و نکست
          </button>
          <button
            class="btn btn-neutral btn-lg "
            type="button"
          >
            دوره های ری اکت و نکست
          </button>
        </div>
        <img
          alt="online class"
          class="grayscale mt-4 opacity-70 m-auto xl:m-0"
          data-nimg="1"
          decoding="async"
          height="39"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75"
          srcset="/_next/image?url=%2Fimages%2Fframeworks.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75 2x"
          style="color: transparent;"
          width="412"
        />
      </div>
      <img
        alt="online class"
        data-nimg="1"
        decoding="async"
        height="521"
        loading="lazy"
        src="/images/programmer-landing.svg"
        style="color: transparent;"
        width="702"
      />
    </div>
  </section>
</div>
`;
